File: backend/components/metiers/administration_publique/plugins/core/advanced_plugin.test.js
// advanced_plugin.test.js – Test JS plugin avancé
const { AdvancedPlugin } = require('./advanced_plugin');
test('AdvancedPlugin workflow', () => {
  const plugin = new AdvancedPlugin();
  expect(() => plugin.run({})).toThrow();
  plugin.activate({ user: 'admin' });
  expect(plugin.activated).toBe(true);
  const res = plugin.run({ foo: 'bar' });
  expect(res).toMatchObject({ foo: 'bar', plugin: 'advanced', status: 'ok' });
  expect(plugin.getAuditTrail().length).toBe(2);
});

File: backend/components/metiers/administration_publique/plugins/core/advanced_plugin.test.py
# advanced_plugin.test.py – Test Python plugin avancé
from .advanced_plugin import AdvancedPlugin
import pytest

def test_advanced_plugin_workflow():
    plugin = AdvancedPlugin()
    with pytest.raises(Exception):
        plugin.run({})
    plugin.activate({'user': 'admin'})
    assert plugin.activated is True
    res = plugin.run({'foo': 'bar'})
    assert res['foo'] == 'bar'
    assert res['plugin'] == 'advanced'
    assert res['status'] == 'ok'
    assert len(plugin.get_audit_trail()) == 2
